node/libs/crypto/src/bn254/tests.rs

```diff
@@ -89,6 +89,18 @@ fn aggregate_signature_failure_smoke() {
     assert!(agg.verify(pks.iter().map(|pk| (&msg[..], pk))).is_err())
 }
 
+#[test]
+fn pk_validity_checks() {
+    // Check that we can't decode the null point.
+    let pk = PublicKey::default();
+    let bytes = pk.encode();
+    let pk_decoded = PublicKey::decode(&bytes);
+    let err = pk_decoded.unwrap_err();
+    assert_eq!(format!("{}", err.root_cause()), "Public key can't be zero");
+
```

Comment:
don't compare human readable error messages. Just use `assert!(pk_decoded.is_err())`. Alternatively implement a decode() method which returns a strongly typed error that you may match against and make ByteFmt::decode just call your custom decode.